ppl/zqd/recruiter/workerreg.go

```diff
@@ -0,0 +1,65 @@
+package recruiter
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/brimsec/zq/api"
+	"github.com/brimsec/zq/api/client"
+	"go.uber.org/zap"
+)
+
+type WorkerReg struct {
+	conn          *client.Connection
+	recruiteraddr string
+	selfaddr      string
+	nodename      string
+}
+
+func NewWorkerReg(ctx context.Context, srvAddr string, recruiteraddr string, specPodIP string, specNodeName string) (*WorkerReg, error) {
```

Comment:
specPodIP AND specNodName deserve better names.

But is either parameter necessary? I don't think there's any scenario in which srvAddr differs from the pod IP address, and the node name can be looked up using the pod IP.

Reply:
Let's address this later.